File: ios-app/dbv-nfc-games/Models/Team.swift
import Foundation

struct Team: Codable, Identifiable {
    let id: UUID
    let gameId: UUID?
    let name: String
    let joinCode: String?
    let color: String
    
    init(id: UUID, gameId: UUID? = nil, name: String, joinCode: String? = nil, color: String) {
        self.id = id
        self.gameId = gameId
        self.name = name
        self.joinCode = joinCode
        self.color = color
    }
}

struct Player: Codable, Identifiable {
    let id: UUID
    let displayName: String
    let deviceId: String
}

File: web/src/types/team.ts
export interface Team {
  id: string
  gameId: string
  name: string
  joinCode: string
  color: string
  /** OW-05: most players this team takes; null or absent means no limit. */
  maxPlayers?: number | null
}

export interface Player {
  id: string
  teamId: string
  deviceId: string
  displayName: string
  /** PF-01: saved to an account. Never the address. */
  hasAccount?: boolean
}
